dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -221,4 +230,71 @@ public KernelBuilder Configure(Action<KernelConfig> configure)
         this._aiServices.SetService<TService>(serviceId, () => factory(this._loggerFactory, this._config), setAsDefault);
         return this;
     }
+
+    /// <summary>
+    /// Create a default prompt template engine.
+    /// </summary>
+    /// <param name="loggerFactory">Logger factory to be used by the template engine</param>
+    /// <returns></returns>
+    private IPromptTemplateEngine CreateDefaultPromptTemplateEngine(ILoggerFactory? loggerFactory = null)
+    {
+        if (!_promptTemplateEngineInit)
+        {
+            _promptTemplateEngineType = this.GetPromptTemplateEngineType();
+            _promptTemplateEngineInit = true;
+        }
+
+        if (_promptTemplateEngineType is not null)
+        {
+            var constructor = _promptTemplateEngineType.GetConstructor(new Type[] { typeof(ILoggerFactory) });
+            if (constructor is not null)
+            {
+#pragma warning disable CS8601 // Null logger factory is OK
+                return (IPromptTemplateEngine)constructor.Invoke(new object[] { loggerFactory });
+            }
+        }
+
+        return new NoopPromptTemplateEngine();
+    }
+
+    /// <summary>
+    /// Get the prompt template engine type if available
+    /// </summary>
+    /// <returns>The type for the prompt template engine if available</returns>
+    private Type? GetPromptTemplateEngineType()
+    {
+        try
+        {
+            var assembly = Assembly.Load("Microsoft.SemanticKernel.TemplateEngine");
+
+            return assembly.ExportedTypes.Single(type =>
+                type.Name.Equals("PromptTemplateEngine", StringComparison.Ordinal) &&
+                type.GetInterface(nameof(IPromptTemplateEngine)) is not null);
+        }
+        catch (FileNotFoundException)
+        {
+            // Unable to load the Microsoft.SemanticKernel.TemplateEngine assembly
+        }
+        catch (InvalidOperationException)
+        {
+            // Assembly does not contain typed named PromptTemplateEngine
+        }
+        catch (Exception ex) when (!ex.IsCriticalException())
+        {
+            // Something unexpected by not critical
+        }
+
+        return null;
+    }
+}
+
+/// <summary>
+/// No-operation IPromptTemplateEngine which performs no rendering of the template.
+/// </summary>
+internal class NoopPromptTemplateEngine : IPromptTemplateEngine
```

Comment:
It's usually a good practice to have a file per class or interface, so consider to move this class declaration to a separete file.

Reply:
Should we consider calling it `NullPromptTemplateEngine` ?

Reply:
+1 for `NullPromptTemplateEngine` :)